mkdocs/nav.py

```diff
@@ -243,14 +243,18 @@ def is_homepage(self):
     def is_top_level(self):
         return len(self.ancestors) == 0
 
-    def load_markdown(self):
-        try:
-            input_content = io.open(self.abs_input_path, 'r', encoding='utf-8').read()
-        except IOError:
-            log.error('file not found: %s', self.abs_input_path)
-            raise
-
-        self.markdown, self.meta = meta.get_data(input_content)
+    def read_source(self, config):
+        source = config['plugins'].run_event(
+            'page_read_source', None, config=config, page=self)
+        if source is None:
+            try:
+                with open(self.abs_input_path, 'r', encoding='utf-8') as f:
```

Comment:
You need to use `io.open` here as was used in the code you are replacing. `io.open` guarantees feature parity between Python2 & 3. And then both the Python2.7 and flake8 tests will pass.